heltour/tournament/views.py

```diff
@@ -1801,6 +1801,54 @@ def view(self, post=False):
     def view_post(self):
         return self.view(post=True)
 
+class ConfirmScheduledTimeView(LeagueView, LoginRequiredMixin):
+    def view(self, post=False):
+        if not self.request.user.is_authenticated:
+            return redirect('by_league:league_home', self.league.tag)
+        player = Player.get_or_create(self.request.user.username)
+
+        active_seasons = self.league.season_set.filter(is_completed=False).order_by('-start_date')
+        active_seasons_with_sp = [(s, player.seasonplayer_set.filter(season=s).first()) for s in
+                                  active_seasons]
+        active_seasons_with_sp = [s for s in active_seasons_with_sp if s[1]]
+        last_sp = player.seasonplayer_set.filter(season__league=self.league, season__is_active=True,
+                                                 season__is_completed=True) \
+            .order_by('-season__start_date').first()
+        last_season = last_sp.season if last_sp is not None else None
+
+        active_rounds = Round.objects.filter(publish_pairings=True, is_completed=False,
+                                             season__is_active=True)
+        next_pairings = []
+        for r in active_rounds:
+            next_pairings += [(r, p) for p in
+                            r.pairings.filter(white=player).exclude(scheduled_time=None) | r.pairings.filter(
+                                black=player).exclude(scheduled_time=None)]
+
+        context = {
+            'player': player,
+            'active_seasons_with_sp': active_seasons_with_sp,
+            'last_season': last_season,
+            'next_pairings': next_pairings
+        }
+
+        if post:
```

Comment:
if this is a class based view, you should be able to overload `get(request)` and `post(request)` independantly.

https://ccbv.co.uk/